doc/source/dev/gitwash/development_workflow.rst

```diff
@@ -195,27 +187,78 @@ Standard acronyms to start the commit message with are::
    REL: related to releasing numpy
 
 
+.. _asking-for-merging:
+
+Asking for your changes to be merged with the main repo
+=======================================================
+
+When you feel your work is finished, you can file a pull request and
+submit an optional code review. Code reviews are typically submitted when an
+API change is proposed, such as adding a new function or modifying an option
+for an existing function.
+
+Github has a help page that outlines the process for `filing pull requests`_.
+If you are going to file a code review, copy the URL for the `pull request
+review`_ between your branch and ``master``, and post it to the `NumPy mailing
+list`_ asking for review. 
+
+.. note:: 
+
+    Make sure your comparison is with ``master`` at the point where you
+    created your working branch, which is indicated by an ellipsis (``...``).
+    The importance of this notation is outlined in :ref:`dot2-dot3`.
+
+
+.. _pushing-to-main:
+
+Pushing changes to the main repo
+================================
+
```

Comment:
Maybe add a note here that this is only relevant if you have commit rights to Numpy itself?